userspace/falco/stats_writer.cpp

```diff
@@ -161,39 +181,220 @@ void stats_writer::worker() noexcept
+	output_fields["falco.version"] = FALCO_VERSION;
+	output_fields["falco.start_ts"] = std::to_string(agent_info->start_ts_epoch);
+	output_fields["falco.duration_sec"] = std::to_string((now - agent_info->start_ts_epoch) / ONE_SECOND_IN_NS);
+	output_fields["falco.kernel_release"] = agent_info->uname_r;
+	output_fields["falco.host_boot_ts"] = std::to_string(machine_info->boot_ts_epoch);
+	output_fields["falco.hostname"] = machine_info->hostname; /* Explicitly add hostname to log msg in case hostname rule output field is disabled. */
+	output_fields["falco.host_num_cpus"] = std::to_string(machine_info->num_cpus);
 
+	output_fields["evt.source"] = src;
+	for (size_t i = 0; i < sizeof(all_driver_engines) / sizeof(const char*); i++)
+	{
+		if (inspector->check_current_engine(all_driver_engines[i]))
+		{
+			output_fields["scap.engine_name"] = all_driver_engines[i];
+			break;
+		}
+	}
+
+	/* Falco userspace event counters. Always enabled. */
+	if (m_last_num_evts != 0 && stats_snapshot_time_delta_sec > 0)
+	{
+		/* Successfully processed userspace event rate. */
+		output_fields["falco.evts_rate_sec"] = std::to_string((num_evts - m_last_num_evts) / (double)stats_snapshot_time_delta_sec);
+	}
+	output_fields["falco.num_evts"] = std::to_string(num_evts);
+	output_fields["falco.num_evts_prev"] = std::to_string(m_last_num_evts);
+	m_last_num_evts = num_evts;
 }
 
-void stats_writer::collector::collect(std::shared_ptr<sinsp> inspector, const std::string& src)
+void stats_writer::collector::get_metrics_output_fields_additional(
+		std::unordered_map<std::string, std::string>& output_fields,
+		const std::shared_ptr<sinsp>& inspector,
+		double stats_snapshot_time_delta_sec, const std::string& src)
 {
-	// just skip if no output is configured
-	if (m_writer->has_output())
+	const scap_agent_info* agent_info = inspector->get_agent_info();
+	const scap_machine_info* machine_info = inspector->get_machine_info();
+
+#ifndef MINIMAL_BUILD
+	/* Resource utilization, CPU and memory usage etc. */
+	uint32_t nstats = 0;
+	int32_t rc = 0;
+	if (m_writer->m_config->m_metrics_resource_utilization_enabled)
 	{
-		// collect stats once per each ticker period
-		auto tick = stats_writer::get_ticker();
-		if (tick != m_last_tick)
+		const scap_stats_v2* utilization;
+		auto buffer = inspector->get_sinsp_stats_v2_buffer();
+		utilization = libsinsp::resource_utilization::get_resource_utilization(agent_info, buffer, &nstats, &rc);
+		if (utilization && rc == 0 && nstats > 0)
 		{
-			stats_writer::msg msg;
-			msg.stop = false;
-			msg.source = src;
-			inspector->get_capture_stats(&msg.stats);
-			m_samples++;
-			if(m_samples == 1)
+			for(uint32_t stat = 0; stat < nstats; stat++)
 			{
-				msg.delta = msg.stats;
+				char metric_name[STATS_NAME_MAX] = "falco.";
+				strncat(metric_name, utilization[stat].name, sizeof(metric_name) - strlen(metric_name));
```

Comment:
```
If src contains n or more bytes, strncat() writes n+1 bytes to
       dest (n from src plus the terminating null byte).  Therefore, the
       size of dest must be at least strlen(dest)+n+1.
```

We should do `sizeof(metric_name) - strlen(metric_name) - 1` here to avoid missing the terminator char.
We can do it in a subsequent PR but please do that :) @jasondellaluce